src/control/server/main.go

```diff
@@ -111,8 +120,10 @@ func serverMain() error {
 	// Create new grpc server, register services and start serving (after
 	// dropping privileges).
 	var sOpts []grpc.ServerOption
-	// TODO: This will need to be extended to take certificate information for
-	// the TLS protected channel. Currently it is an "insecure" channel.
+
+	if creds != nil {
```

Comment:
This test seems a little weird. Can creds ever be nil if err was nil? If so, that's probably something to be tested in the functions which load the credentials, such that creds is never nil if err is nil.

Reply:
Yes it can. If we say we're using insecure channels we will have a nil creds passed in and it will mean we don't add the transport credentials to the grpc server options.

Reply:
I can see that case (creds could be nil) in control.go, but here, your creds is being returned from security.LoadServerCreds(), which as far as I can see should never return a nil creds if err is nil. It's a minor point, but I'm just concerned that this might mask an error which should result in a noisy failure.

Reply:
Ahh ok yea in that case that was correct. It was that way because I hadn't implemented the insecure flag in the server with the config file at that point. The new version there is a check for insecure at the beginning of a loadCredentials call and it can return nil,nil.